src/NServiceBus.Core/Routing/AutomaticSubscriptions/AutoSubscribe.cs

```diff
@@ -56,20 +59,28 @@ static List<Type> GetMessageTypesHandledByThisEndpoint(MessageHandlerRegistry ha
 
         class ApplySubscriptions : FeatureStartupTask
         {
-            public ApplySubscriptions(IEnumerable<Type> messagesHandledByThisEndpoint, Func<Type, Task<bool>> asyncPredicate)
+            public ApplySubscriptions(IEnumerable<Type> messagesHandledByThisEndpoint, Func<Type, bool> predicate)
             {
                 this.messagesHandledByThisEndpoint = messagesHandledByThisEndpoint;
-                this.asyncPredicate = asyncPredicate;
+                this.predicate = predicate;
             }
 
             protected override async Task OnStart(IMessageSession session)
             {
                 foreach (var eventType in messagesHandledByThisEndpoint)
                 {
-                    if (await asyncPredicate(eventType).ConfigureAwait(false))
+                    if (predicate(eventType))
                     {
-                        await session.Subscribe(eventType).ConfigureAwait(false);
-                        Logger.DebugFormat("Auto subscribed to event {0}", eventType);
+                        try
+                        {
+                            await session.Subscribe(eventType).ConfigureAwait(false);
+                            Logger.DebugFormat("Auto subscribed to event {0}", eventType);
+                        }
+                        catch (Exception ex)
+                        {
+                            throw new Exception("s", ex);
```

Comment:
I was testing if I really get an exception here before I realized it is getting swallowed by the testing framework. Forgot to remove this code. Corrected now.